pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -33,16 +33,27 @@ import (
 )
 
 const (
-	OrchestrationStatusStarting      = "starting"
+	// OrchestrationStatusStarting denotes the OSD provisioning is beginning.
+	OrchestrationStatusStarting = "starting"
+	// OrchestrationStatusComputingDiff denotes the OSD provisioning is scanning for disks that can
+	// be provisioned.
 	OrchestrationStatusComputingDiff = "computingDiff"
+	// OrchestrationStatusOrchestrating denotes the OSD provisioning has begun and is running.
 	OrchestrationStatusOrchestrating = "orchestrating"
-	OrchestrationStatusCompleted     = "completed"
-	OrchestrationStatusFailed        = "failed"
-	orchestrationStatusMapName       = "rook-ceph-osd-%s-status"
-	orchestrationStatusKey           = "status"
-	provisioningLabelKey             = "provisioning"
-	nodeLabelKey                     = "node"
-	completeProvisionTimeout         = 20
+	// OrchestrationStatusCompleted denotes the OSD provisioning has completed. This does not imply
+	// the provisioning completed successfully in whole or in part.
+	OrchestrationStatusCompleted = "completed"
+	// OrchestrationStatusPreexisting denotes the OSD provisioning was not started because an OSD
+	// has already been provisioned and started.
+	OrchestrationStatusPreexisting = "preexisting"
```

Comment:
Shouldn't this be "pending", because, in the end, this is what it is doing right? It has been deferred and waiting for something else to complete so then it can proceed.

Reply:
Umm... I think this is specifically talking about OSD provisioning, which I wouldn't conceptually think of as "pending". Provisioning has already completed, but we need to be able to tell the difference between _provisioning completed and created a new OSD_ ("completed") and _provisioning already happened on a prior reconcile and was never started_ (what I call "preexisting"). I wish there were a better name than "preexisting", but I can't think of one. Though "pending" doesn't make sense to me for this case.

Reply:
what about `alreadyProvisioned`?